src/bosh_azure_cpi/lib/cloud/azure/restapi/azure_client.rb

```diff
@@ -383,6 +356,17 @@ def create_virtual_machine(resource_group_name, vm_params, network_interfaces, a
         end
       end
 
+      unless vm_params[:config_disk].nil?
+        # append the config disk into the vm.
+        vm['properties']['storageProfile']['dataDisks'].push(
+          'lun' => 1,
+          'createOption' => 'Attach',
+          'managedDisk' => {
```

Comment:
config disks are only supported for VM with managed disks? how about unmanaged disk VM?

Reply:
no support for the unmanaged disk for now, and may think about to add the support later.

Reply:
please raise error in vm_manager when use_config_disk is true and use_managed_disk is false.